### PR TITLE
Cache busting

### PR DESCRIPTION
**Pull request by gestchild**
## What is this PR trying to achieve?

Improve performance by telling the browser it's ok to use cached resources and having a way of busting the cache when necessary

## What does it look like?

cached

### CODE REVIEW

File: client/gulpfile.js [hunk 1]
```diff
@@ -98,8 +134,8 @@ gulp.task('js:lint', () => {
 });
 
 gulp.task('watch', () => {
-  gulp.watch(sources.scss.all, ['scss:compile']);
-  gulp.watch(sources.js.all, ['js:compile']);
+  gulp.watch(sources.scss.all, ['bustCSS']);
```
Comment (jamesgorrie) [inline client/gulpfile.js]: Could we use a similar naming as in `bust:css|js`?

**Comment by jamesgorrie** [comment]
Could you post what those config files look like here please?

**Comment by gestchild** [comment]
{"application.css":"application-fb52858316045e1e.css"}

File: client/gulpfile.js [hunk 2]
```diff
@@ -53,8 +58,19 @@ gulp.task('icons:copy', () => {
...
 });
 
+// TODO move paths to vars and synchronous stuff
+gulp.task('cleanCSS', function () {
+  return gulp.src('../dist/assets/css/')
+    .pipe(clean({force: true}));
+});
+
+gulp.task('cleanJS', function () {
+  return gulp.src('../dist/assets/js/')
```
Comment (jamesgorrie) [inline client/gulpfile.js]: Isn't this a reference to [this](https://github.com/wellcometrust/wellcomecollection.org/pull/457/files#diff-64bf274ee0d4e9c521933759c5e52d8fL24)?

File: server/package.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "main": "app.js",
   "scripts": {
     "app:build": "babel . -d .dist --ignore=node_modules",
-    "app:dev": "nodemon app.js -e js,njk --exec babel-node",
+    "app:dev": "nodemon --watch app.js --watch ../dist/assets/js/app.js --watch ../dist/assets/css/application.css -e js,njk,css --exec babel-node app.js",
```
Comment (jamesgorrie) [inline server/package.json]: Could we rather just watch the JSON files?

File: server/middleware/set-cache-control.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+export default (options) => {
+  options = options || {};
```
Comment (jamesgorrie) [inline server/middleware/set-cache-control.js]: To avoid mutation you could use `options = {}` in the arguments.

File: server/package.json (hunk 3 above)
Comment (gestchild) [inline server/package.json]: That makes more sense, yep

**Pull request merged**